### DISCUSSION TITLE
Apple // AppleWorks 3.0 ADB file

### DISCUSSION THREAD

**Discussion by bachelorbob** [fadden/CiderPress2#18] [Q&A] [answered]
I have gotten to the point of having an Apple // AppleWorks 3.0 ADB file on my Mac's HD.  It had no file extension, so I added ADB.  I can view it as a text file with garbage formatting.  I have the GUI interface of CiderPress II 1.0.1 running in Windows 10 via Parallels Desktop.

CiderPress II gives "Error: unable to recognize contents of 'DATABASE.ADB': file extension not a known disk or file archive (ext='.adb')"

What should the extension be?  I just need to convert this 'ancient' database file to something that a current application can read, such as a CSV, TSV, etc.

Same goes for CiderPress v4.1.1

**Comment by fadden** [comment, marked as answer]
ProDOS files have a file type and an auxiliary type, both of which are lost when the file is stored on a modern filesystem. The CiderPress file converters are intended to operate on files stored in a format that preserves the file attributes; this could be a disk image, file archive, or wrapper like AppleSingle/AppleDouble.

The file needs to be put back into a disk image or file archive, with the correct file type, so that the file converters will be able to figure out what it is.  AppleWorks ADB use file type $19, with an auxtype that doesn't really matter because it's used to format the filename (to get spaces and mixed case on ProDOS).  We can add a NAPS preservation string to set the appropriate type information.

CLI (can be done on the Mac):
```
% mv MyFile.ADB MyFile#190000
% cp2 create-file-archive mystuff.shk
% cp2 add mystuff.shk MyFile#190000
Adding 1 file
Adding MyFile#190000 -> MyFile
% cp2 export mystuff.shk adb
exporting MyFile -> MyFile.csv [adb]
```

GUI:
- Rename "MyFile.ADB" to "MyFile#190000".
- Use File > New File Archive to create a new ShrinkIt (NuFX) archive.
- Drag & drop MyFile#190000 into the archive, with "Add/Extract" set in the toolbar (NAPS must be checked in the Add/Import options on the right side; it should be by default).  You should see a file called "MyFile" with type ADB and auxtype $0000, appear in the file list.
- (optional) Double-click on the entry to view it as CSV.
- Click "Import/Export" in the toolbar, and drag MyFile out of the archive to create the .csv file.

Note the original CiderPress has a bug in the file conversion: https://github.com/fadden/ciderpress/issues/66

**Reply by bachelorbob** [reply]
Perfect.  And just like that, a 1994, 30-year-old Apple //c database is back to life.

Many, many thanks.
